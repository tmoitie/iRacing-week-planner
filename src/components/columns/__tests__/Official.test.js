import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, test } from '@jest/globals';
import { TableWrapper } from './ColumnUtils';
import '@testing-library/jest-dom';

import Official from '../Official';

describe('components/columns/Official', () => {
  test('renders true', () => {
    render(<TableWrapper><Official race={{ official: true }} /></TableWrapper>);

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();
    expect(screen.queryByTestId('TickIcon')).toBeInTheDocument();
  });

  test('renders false', () => {
    render(<TableWrapper><Official race={{ official: false }} /></TableWrapper>);

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();
    expect(screen.queryByTestId('TickIcon')).not.toBeInTheDocument();
  });
});
