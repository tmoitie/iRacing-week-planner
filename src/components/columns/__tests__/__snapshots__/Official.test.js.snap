// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/Official renders false 1`] = `
<td>
  <div />
</td>
`;

exports[`components/columns/Official renders true 1`] = `
<td>
  <div>
    <span
      class="glyphicon glyphicon-ok"
      data-testid="TickIcon"
    />
  </div>
</td>
`;
